src/database/accessors/loadProposalBundle.ts

```diff
@@ -1,16 +1,222 @@
+);
+const isProposalPiece = (
+  o: Record<string, unknown>,
+): o is ProposalPiece => (
+  'proposal_id' in o
+);
+
+const extractApplicationFormFields = (
+  applicationFormFieldPieces: ApplicationFormFieldPiece[],
+): ApplicationFormField[] => (
+  applicationFormFieldPieces.reduce<ApplicationFormField[]>(
+    (applicationFormFields, applicationFormFieldPiece) => [
+      ...applicationFormFields,
+      {
+        id: applicationFormFieldPiece.applicationFormField_id,
+        applicationFormId: applicationFormFieldPiece.applicationFormField_applicationFormId,
+        canonicalFieldId: applicationFormFieldPiece.applicationFormField_canonicalFieldId,
+        position: applicationFormFieldPiece.applicationFormField_position,
+        label: applicationFormFieldPiece.applicationFormField_label,
+        createdAt: applicationFormFieldPiece.applicationFormField_createdAt,
+      },
+    ],
+    [],
+  ).filter(filterByDistinctId)
+);
+
+const extractProposalFieldValues = (
+  proposalFieldValuePieces: ProposalFieldValuePiece[],
+): ProposalFieldValue[] => (
+  proposalFieldValuePieces.reduce<ProposalFieldValue[]>(
+    (proposalFieldValues, proposalFieldValuePiece) => [
+      ...proposalFieldValues,
+      {
+        id: proposalFieldValuePiece.proposalFieldValue_id,
+        proposalVersionId: proposalFieldValuePiece.proposalFieldValue_proposalVersionId,
+        applicationFormFieldId: proposalFieldValuePiece.proposalFieldValue_applicationFormFieldId,
+        position: proposalFieldValuePiece.proposalFieldValue_position,
+        value: proposalFieldValuePiece.proposalFieldValue_value,
+        createdAt: proposalFieldValuePiece.proposalFieldValue_createdAt,
+      },
+    ],
+    [],
+  ).filter(filterByDistinctId)
+);
+
+const extractProposalVersions = (
+  proposalVersionPieces: ProposalVersionPiece[],
+): ProposalVersion[] => (
+  proposalVersionPieces.reduce<ProposalVersion[]>(
+    (proposalVersions, proposalVersionPiece) => [
+      ...proposalVersions,
+      {
+        id: proposalVersionPiece.proposalVersion_id,
+        proposalId: proposalVersionPiece.proposalVersion_proposalId,
+        applicationFormId: proposalVersionPiece.proposalVersion_applicationFormId,
+        version: proposalVersionPiece.proposalVersion_version,
+        createdAt: proposalVersionPiece.proposalVersion_createdAt,
+      },
+    ],
+    [],
+  ).filter(filterByDistinctId)
+);
+
+const extractProposals = (
+  proposalPieces: ProposalPiece[],
+): Proposal[] => (
+  proposalPieces.reduce<Proposal[]>(
+    (proposals, proposalPiece) => [
+      ...proposals,
+      {
+        id: proposalPiece.proposal_id,
+        applicantId: proposalPiece.proposal_applicantId,
+        opportunityId: proposalPiece.proposal_opportunityId,
+        externalId: proposalPiece.proposal_externalId,
+        createdAt: proposalPiece.proposal_createdAt,
+      },
+    ],
+    [],
+  ).filter(filterByDistinctId)
+);
+
+const joinProposalFieldValuesWithApplicationFormFields = (
```

Comment:
Similar to the piece extraction: these joiners (which maybe we could / should still call `enrichers`) belong in their own directory as well.